### PR TITLE
Simple Payments: Remove feature flag

### LINKED ISSUE #25326: Simple Payments: remove feature flag in Calypso
Let's remove the 'simple-payments' feature flag in Calypso, this is currently enabled for Premium and Business plan users. I can't think of any use case of where we'd disable this now, so let's tidy a bit.

### PR DESCRIPTION
**Pull request by Copons**
Fix #25326

Remove the `simple-payments` feature flag from all environments, and all its uses across Calypso.

Also, take care of some a11y concerns in the HTML Editor toolbar's Add Content dropdown: replace some `div` with click events with `button` so that the commit does not blow up several ESLint's `jsx/a11y` rules.

## Testing instructions

- Make sure that the Simple Payments dialog still works in both Visual and HTML editor for Premium and Business sites.
- Make sure Simple Payments are not enabled for Free and Personal sites.
- Additionally: check that the Add Content dropdown in the HTML toolbar still works the same as the one in the Visual toolbar.

### CODE REVIEW

File: client/post-editor/editor-html-toolbar/index.jsx [hunk 1]
```diff
@@ -509,51 +509,49 @@ export class EditorHtmlToolbar extends Component {
 
 		return (
 			<div className={ insertContentClasses }>
-				<div
+				<button
```
Comment (gwwar) [inline client/post-editor/editor-html-toolbar/index.jsx]: Let's save this refactor for a different PR

**Comment by gwwar** [comment]
Code looks good, need a rebase and some manual testing 👍 . Only area where this wasn't enabled was in Desktop, though I can't think of why that wasn't enabled except that we forgot to do so.

File: client/post-editor/editor-html-toolbar/index.jsx (hunk 1 above)
Comment (Copons) [reply client/post-editor/editor-html-toolbar/index.jsx]: This refactor is needed to clear the precommit a11y lint though, and it only requires 2 lines of CSS to look the same.

Comment (gwwar) [reply client/post-editor/editor-html-toolbar/index.jsx]: ah gotcha 👍 I missed that in the summary.

**Pull request merged**